src/library/scala/collection/mutable/LinkedHashSet.scala

```diff
@@ -95,6 +95,64 @@ class LinkedHashSet[A]
     }
   }
 
+  override def filterInPlace(p: A => Boolean): LinkedHashSet.this.type = {
+    if (size > 0) {
+      if (size < 1000000) {
```

Comment:
When you improve the `LinkedHashMap` code, I bet the 1M thing will go away.  So, probably, you'll just need to redo this all to match LHM.